### DISCUSSION TITLE
PV_VENV or --venv not finding python environment

### DISCUSSION THREAD

**Discussion by earlsioson** [Kitware/trame#64] [Q&A] [answered]
Hello,

We are trying to build a docker image with the ParaView-5.10.1-osmesa-MPI-Linux-Python3.9-x86_64.tar.gz install. We've followed the instructions in the examples/ParaView/README.md but the example is not finding the python environment set with either the PV_VENV or --venv options.

<img width="1439" alt="image" src="https://user-images.githubusercontent.com/13466524/165750974-ad3ca560-d6ac-4b2d-997e-1fb0705fe78e.png">

**Comment by jourdain** [comment, marked as answer]
Are you sure that your path `~/.pvenv` is correct and point to a venv using the same python version of paraview and where trame was install?

**Reply by earlsioson** [reply]
@jourdain, yes pip list, when activating ~/.pvenv shows trame and both pvpython and python are at 3.9.5. I did notice that although they are both at 3.9.5 they are built with different versions of GCC on different Linux versions. Would this be the reason why it is not working?
<img width="782" alt="image" src="https://user-images.githubusercontent.com/13466524/165939860-4f18eb54-4b10-4e2c-93e4-edd6faf95dba.png">

**Reply by jourdain** [reply]
If your python packages inside your venv don't link to Python, it won't be an issue which is usually the case.

Can you see if the code in  `paraview.web.venv` get's executed and see why it is not "activating" your venv...

If you are in docker be aware that venv are not relocatable, so creating a venv on the host with a different path than the one in docker won't work...

**Reply by jourdain** [reply]
Any update on that topic? Did you find the issue?

**Reply by earlsioson** [reply]
@jourdain, no update, I wasn't able to figure it out. Not sure how I would use paraview.web.venv to diagnose the problem.

**Reply by jourdain** [reply]
since it is Python you can directly edit that file within ParaView and see why it is not loading the venv for you...

**Reply by earlsioson** [reply]
@jourdain, I tried this again using the latest kiteware/trame:glvnd image as a base image and it's working now.
